.github/workflows/build-check-install-on-macos.yaml

```diff
@@ -0,0 +1,729 @@
+        default: ""
+        type: string
+      additional-r-cmd-check-params:
+        description: Additional flags or parameters to add to R CMD check
+        required: false
+        default: ""
+        type: string
+      disable-unit-test-reports:
+        description: Do not produce unit test reports
+        required: false
+        default: false
+        type: boolean
+      additional-env-vars:
+        description: |
+          Extra environment variables, as a 'key=value' pair, with each pair on a new line.
+          Example usage:
+          additional-env-vars: |
+            ABC=123
+            XYZ=456
+        required: false
+        default: ""
+        type: string
+      junit-xml-storage:
+        description: Branch name to store JUnit XML reports
+        required: false
+        default: "_junit_xml_reports"
+        type: string
+      junit-xml-diff-branch:
+        description: Against which branch should the JUnit XML report from the current branch be compared?
+        required: false
+        default: main
+        type: string
+      junit-xml-comparison:
+        description: Turn on JUnit XML comparison
+        required: false
+        default: true
+        type: boolean
+      junit-xml-negative-threshold:
+        description: |
+          If time difference during JUnit XML comparison is between negative-threshold and 0: it's treated as 0.
+          This means that if test suite/case executes up to negative-threshold seconds faster,
+          this time difference is ignored as insignificant, and test suite/case is not shown in the table
+        required: false
+        default: 1.0
+        type: string
+      junit-xml-positive-threshold:
+        description: |
+          If time difference during JUnit XML comparison is between 0 and positive-threshold: it's treated as 0.
+          This means that if test suite/case executes up to positive-threshold seconds slower,
+          this time difference is ignored as insignificant, and test suite/case is not shown in the table
+        required: false
+        default: 1.0
+        type: string
+      sd-direction:
+        description: The direction to use to install staged dependencies. Choose between 'upstream', 'downstream' and 'all'
+        required: false
+        type: string
+        default: upstream
+      unit-test-report-brand:
+        description: Image URL to use in unit test report for branding. If empty, the default xunit-viewer brand will be used.
+        required: false
+        type: string
+        default: ""
+      publish-unit-test-report-gh-pages:
+        description: Publish HTML unit test report to GitHub Pages alongside pkgdown docs.
+        required: false
+        type: boolean
+        default: true
+
+concurrency:
+  group: r-cmd-on-macos-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-install-check:
+    name: R cmd check on MacOs-latest
+    runs-on: macos-latest
```

Comment:
I've just checked that it is possible to parametrize this with an input. If this was an input, a user could have multiple workflow calls of this workflow, each with different `runs-on` `input`: for example one with `ubuntu-latest` and another one with `macos-latest`.

This is just a suggestion to potentially avoid duplication of code between this workflow and `build-check-install.yaml`.